File: length-of-last-word.go
func lengthOfLastWord(s string) int {
	wordLength := 0
	for i := len(s) - 1; i > -1; i-- {
		if s[i] == ' ' {
			if wordLength > 0 {
				return wordLength
			}
			continue
		}
		wordLength++
		if i == 0 {
			return wordLength
		}
	}
	return -1
}

File: length-of-last-word.c
int lengthOfLastWord(char *s) {
    int wordLength = 0;
    int sLen = (int)strlen(s) - 1;
    for (int i = sLen; i > -1; i--) {
        if (s[i] == ' ') {
            if (wordLength > 0)
                return wordLength;
            continue;
        }
        wordLength++;
        if (i == 0)
            return wordLength;
    }
    return -1;
}
